exporters/otlp/otlpmetric/otlpmetrichttp/client.go

```diff
@@ -189,11 +192,21 @@ func (c *client) UploadMetrics(ctx context.Context, protoMetrics *metricpb.Resou
 			// Retry-able failure.
 			rErr = newResponseError(resp.Header)
 
-			// Going to retry, drain the body to reuse the connection.
-			if _, err := io.Copy(io.Discard, resp.Body); err != nil {
-				_ = resp.Body.Close()
+			// In all these cases, since the response body is
+			// not getting parsed, actual reason of failure is
+			// very hard to find out for the user.
+			// If body is not empty, then we can use the msg
+			// of underlying service as error message.
```

Comment:
```suggestion
```

This comment makes sense within the PR, but outside its context, it doesn't that much.

Reply:
Addressed. Let me know if it needs any improvement